Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/ExpanderSettings.cpp

```diff
@@ -0,0 +1,201 @@
+        const AZ::DocumentPropertyEditor::DocumentAdapter* adapter,
+        const AZStd::string& settingsRegistryKey,
+        const AZStd::string& propertyEditorName)
+        : m_shouldSettingsPersist(true)
+        , m_adapter(adapter)
+    {
+        if (!settingsRegistryKey.empty() && !propertyEditorName.empty())
+        {
+            m_settingsRegistryBasePath = AZStd::string::format("%s/%s", RootSettingsRegistryPath, propertyEditorName.c_str());
+            m_fullSettingsRegistryPath = AZStd::string::format("%s/%s", m_settingsRegistryBasePath.c_str(), settingsRegistryKey.c_str());
+
+            m_settingsFilepath.Append(RootSettingsFilepath)
+                .Append(AZStd::string::format("%s_settings", propertyEditorName.c_str()))
+                .ReplaceExtension(SettingsRegistrar::SettingsRegistryFileExt);
+
+            m_wereSettingsLoaded = LoadExpanderStates();
+
+            if (m_wereSettingsLoaded)
+            {
+                SetCleanExpanderStateCallback(
+                [this](ExpanderSettings::ExpanderStateMap& storedStates)
+                {
+                    const auto& rootValue = m_adapter->GetContents();
+                    auto numErased = AZStd::erase_if(
+                        storedStates,
+                        [&rootValue](const AZStd::pair<AZStd::string, bool>& statePair)
+                        {
+                            return !rootValue.FindChild(AZ::Dom::Path(statePair.first)) ? true : false;
+                        });
+                    return numErased > 0;
+                });
+            }
+        }
+    }
+
+    ExpanderSettings::~ExpanderSettings()
+    {
+        SaveAndCleanExpanderStates();
+    }
+
+    void ExpanderSettings::Reflect(AZ::ReflectContext* context)
+    {
+        AZ::SerializeContext* serializeContext = azrtti_cast<AZ::SerializeContext*>(context);
+        if (serializeContext)
+        {
+            serializeContext->Class<ExpanderSettings>()->Version(0)->Field(
+                "ExpandedElements", &ExpanderSettings::m_expandedElementStates);
+        }
+    }
+
+    ExpanderSettings::PathType ExpanderSettings::GetStringPathForDomPath(const AZ::Dom::Path& rowPath) const
+    {
+        return PathType(rowPath.ToString(), AZ::IO::PosixPathSeparator);
+    }
+
+    void ExpanderSettings::SaveExpanderStates()
+    {
+        m_settingsRegistrar.StoreObjectSettings(m_fullSettingsRegistryPath, this);
+
+        // Configuration that the dumper util will use when collecting our data from the SettingsRegistry
+        AZ::SettingsRegistryMergeUtils::DumperSettings dumperSettings;
+        dumperSettings.m_prettifyOutput = true;
+        dumperSettings.m_includeFilter = [pathFilter = m_settingsRegistryBasePath](AZStd::string_view path)
+        {
+            return AZ::SettingsRegistryMergeUtils::IsPathAncestorDescendantOrEqual(pathFilter, path);
+        };
+        dumperSettings.m_jsonPointerPrefix = m_settingsRegistryBasePath;
```

Comment:
The include filter isn't needed as the the key parameter passed to the SaveSettingsToFile function below will only save the settings that are at or below the `m_settingsRegistryBasePath` anchor field.

That is detailed in the Output the Settings Registry documentation: https://www.docs.o3de.org/docs/user-guide/settings/output-settings-registry/#storing-editor-preferences-using-a-key-and-an-anchor

Reply:
fixed, thanks!